src/app/loadout-builder/LoadoutBuilderContainer.tsx

```diff
@@ -0,0 +1,345 @@
+
+export interface LBState {
+  lockedMap: LockedMap;
+  lockedSeasonalMods: LockedModBase[];
+  lockedArmor2Mods: LockedArmor2ModMap;
+  selectedStoreId?: string;
+  statFilters: Readonly<{ [statType in StatTypes]: MinMaxIgnored }>;
+  minimumPower: number;
+  query: string;
+  statOrder: StatTypes[];
+  assumeMasterwork: boolean;
+}
+
+const init = ({
+  stores,
+  location,
+}: {
+  stores: DimStore[];
+  location: Location<{
+    loadout?: Loadout | undefined;
+  }>;
+}): LBState => {
+  let lockedMap: LockedMap = {};
+
+  if (stores.length && location.state?.loadout) {
+    for (const loadoutItem of location.state.loadout.items) {
+      if (loadoutItem.equipped) {
+        const item = getItemAcrossStores(stores, loadoutItem);
+        if (item && isLoadoutBuilderItem(item)) {
+          lockedMap = {
+            ...lockedMap,
+            [item.bucket.hash]: addLockedItem(
+              { type: 'item', item, bucket: item.bucket },
+              lockedMap[item.bucket.hash]
+            ),
+          };
+        }
+      }
+    }
+  }
+  return {
+    lockedMap,
+    statFilters: {
+      Mobility: { min: 0, max: 10, ignored: false },
+      Resilience: { min: 0, max: 10, ignored: false },
+      Recovery: { min: 0, max: 10, ignored: false },
+      Discipline: { min: 0, max: 10, ignored: false },
+      Intellect: { min: 0, max: 10, ignored: false },
+      Strength: { min: 0, max: 10, ignored: false },
+    },
+    lockedSeasonalMods: [],
+    lockedArmor2Mods: {
+      [ModPickerCategories.general]: [],
+      [ModPickerCategories.helmet]: [],
+      [ModPickerCategories.gauntlets]: [],
+      [ModPickerCategories.chest]: [],
+      [ModPickerCategories.leg]: [],
+      [ModPickerCategories.classitem]: [],
+      [ModPickerCategories.seasonal]: [],
+    },
+    minimumPower: 750,
+    query: '',
+    statOrder: statKeys,
+    selectedStoreId: getCurrentStore(stores)?.id,
+    assumeMasterwork: false,
+  };
+};
+
+export type LoadoutBuilderAction =
+  | { type: 'changeCharacter'; storeId: string }
+  | { type: 'statFiltersChanged'; statFilters: LBState['statFilters'] }
+  | { type: 'minimumPowerChanged'; minimumPower: number }
+  | { type: 'queryChanged'; query: string }
+  | { type: 'statOrderChanged'; statOrder: StatTypes[] }
+  | { type: 'lockedMapChanged'; lockedMap: LockedMap }
+  | { type: 'addItemToLockedMap'; item: LockedItemType }
+  | { type: 'removeItemFromLockedMap'; item: LockedItemType }
+  | { type: 'lockedSeasonalModsChanged'; lockedSeasonalMods: LockedModBase[] }
+  | {
+      type: 'lockedMapAndSeasonalModsChanged';
+      lockedMap: LockedMap;
+      lockedSeasonalMods: LockedModBase[];
+    }
+  | { type: 'lockedArmor2ModsChanged'; lockedArmor2Mods: LockedArmor2ModMap }
+  | { type: 'assumeMasterworkChanged'; assumeMasterwork: boolean };
+
+// TODO: Move more logic inside the reducer
+function stateReducer(state: LBState, action: LoadoutBuilderAction): LBState {
```

Comment:
Yeah I think it might actually solve problems in my config PR too. I have some opinions on where this should be now but I will address that in a comment on the controller.